src/phoenix/trace/llama_index/callback.py

```diff
@@ -304,6 +294,23 @@ def _add_spans_to_tracer(
         parent_span_id, event_id = parent_child_id_stack.pop()
         event_data = event_id_to_event_data[event_id]
         event_type = event_data["event_type"]
+        attributes = event_data["attributes"]
+        if event_type is CBEventType.LLM and parent_child_id_stack:
+            preceding_parent_span_id, preceding_event_id = parent_child_id_stack[-1]
+            preceding_event_data = event_id_to_event_data[preceding_event_id]
+            preceding_span_shares_parent_id = preceding_parent_span_id == parent_span_id
+            if (
+                preceding_event_data["event_type"] is CBEventType.TEMPLATING
+                and preceding_span_shares_parent_id
+                and (payload := preceding_event_data["start_event"].payload)
```

Comment:
If payload is empty then we're not popping them? As a rule of thumb, I would avoid compound `if` statements because they tend to hide bugs and stick only to singletons as much as possible (as shown in my suggestions).

Reply:
addressed in [d6f34bd](https://github.com/Arize-ai/phoenix/pull/1475/commits/d6f34bd22154926c9e8593aff4970452b0896042)